### ISSUE TITLE
NEST 2FA changes

### ISSUE THREAD

**Issue by logname** [chrisjshull/homebridge-nest#179]
Google seems to have changed their two factor authentication technique. I now cannot enter the 2FA code in my config.json file on version 3.5.1, because it generates a new 2FA code immediately when starting Homebridge again. No more delay before a new code is generated like before.

**Comment by adriancable** [comment]
@logname - this is very interesting. Right now there are 3 ways to use Nest 2FA:

1. Enter the 2FA code in the config.json.
2. Don't enter the 2FA code in the config.json - homebridge will prompt for it on start-up.
3. Use the generateNestToken.sh script to make an access token, and put that in your config.json.

I believe that #2 and #3 will still work, even if Nest's change means #1 no longer works. Can you confirm?

(FWIW, I think that Nest allowing multiple 2FA requests while producing the same code was not a good idea, even though it was helpful for the plug-in.)

**Comment by logname** [comment]
Method #2 still works. Thanks. I had forgotten about that. I did not choose the token method when first setting up because you recommended against it in the readme, for what sounded like very valid reasons to me.

There's another twist. Family sharing is also no longer possible for new users. Not sure if you were aware of this. I spent some time on chat with Google today and they confirmed that if the user had an existing Nest account, they could be invited as a family member. But if you attempt to invite a new user (as I was trying to create just for this purpose only), they will have a Google Nest account, and cannot be invited to your grandfathered Nest account. Only way you can invite a new user is to upgrade to a Google Nest account yourself. Obviously I'm not going to do that until the dust settles.

Thanks for the great plug-in. Has helped me with a bridge to still be able to tie lights to my Nest Protects until google gets their act together.

**Comment by adriancable** [comment]
@logname - it is probably too late to do this now, but before new Nest accounts were discontinued I set up family sharing, inviting email address B to my main account A. Then, once Google accounts came in, I migrated account B to Google. That way, I have two functionally identical accounts with the same devices, but one is a legacy Nest account and one is a Google account. Extremely useful for testing.
